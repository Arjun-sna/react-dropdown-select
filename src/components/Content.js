import React from 'react';
import styled from '@emotion/styled';

import Option from './Option';
import Input from './Input';

const Content = ({ parentProps, parentState, parentMethods }) => {
  return (
    <ContentComponent
      className={`react-dropdown-select-content ${
        parentProps.multi ? 'react-dropdown-select-type-multi' : 'react-dropdown-select-type-single'
      }`}
      onClick={() => parentMethods.dropDown('open')}>
      {parentProps.contentRenderer ? (
        parentProps.contentRenderer(parentProps, parentState, parentMethods)
      ) : (
        <React.Fragment>
          {parentProps.multi
            ? parentState.values &&
              parentState.values.map((item) => (
                <Option
                  key={`${item[parentProps.valueField]}${item[parentProps.labelField]}`}
                  parentItem={item}
                  parentState={parentState}
                  parentProps={parentProps}
                  parentMethods={parentMethods}
                />
              ))
            : parentState.values &&
              parentState.values.length > 0 && <span>{parentState.values[0][parentProps.labelField]}</span>}
          <Input
            parentProps={parentProps}
            parentMethods={parentMethods}
            parentState={parentState}
          />
        </React.Fragment>
      )}
    </ContentComponent>
  );
};

const ContentComponent = styled.div`
  display: flex;
  flex: 1;
  flex-wrap: wrap;
`;

Content.propTypes = {};

export default Content;
