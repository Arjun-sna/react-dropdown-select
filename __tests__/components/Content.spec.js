import React from 'react';
import TestRenderer from 'react-test-renderer';

import Content from '../../src/components/Content';
import { options } from '../options';

let spy;

const props = (props = {}) => ({
  props: {
    contentRenderer: null,
    multi: true,
    labelField: 'name'
  },
  state: {
    search: '',
    values: [options[0]]
  },
  methods: {
    dropDown: () => undefined,
    getInputSize: () => undefined,
  },
  ...props
});

describe('<Clear/> component', () => {
  beforeEach(() => {
    spy = jest.fn();
  });

  afterEach(() => {
    spy = null;
  });

  it('<Content/> renders correctly', () => {
    const tree = TestRenderer.create(<Content {...props()} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('onClick opens dropdown', () => {
    TestRenderer.create(
      <Content {...props()} onClick={spy}/>)
      .root.findByProps({ className: 'react-dropdown-select-content react-dropdown-select-type-multi' }).props.onClick();

    expect(spy).toHaveBeenCalled;
  });
});
