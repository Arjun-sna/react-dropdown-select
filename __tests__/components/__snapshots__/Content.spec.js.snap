// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clear/> component <Content/> renders correctly 1`] = `
<div
  className="react-dropdown-select-content react-dropdown-select-type-multi css-jznujr-ContentComponent e1gn6jc30"
  onClick={[Function]}
>
  <span
    className="react-dropdown-select-option css-1sb8fud-OptionComponent e1l4eby50"
    role="listitem"
  >
    <span
      className="react-dropdown-select-option-label"
    >
      Leanne Graham
    </span>
    <span
      className="react-dropdown-select-option-remove"
      onClick={[Function]}
    >
      Ã—
    </span>
  </span>
  <input
    className="react-dropdown-select-input css-n2ezd1-InputComponent e11wid6y0"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    readOnly={true}
    tabIndex="-1"
    value=""
  />
</div>
`;
